packages/zoe/README.md

```diff
@@ -51,3 +51,60 @@ agoric follow :published.priceFeed.ATOM-USD_price_feed
 ```
 
  TODO document more of https://github.com/Agoric/documentation/issues/672
+
+## Upgrade
+
+A contract instance can be upgraded to use a new source code bundle in a process that is very similar to
+[upgrading a vat](https://github.com/Agoric/agoric-sdk/blob/master/packages/SwingSet/docs/vat-upgrade.md).
+Most state is discarded, however "durable" collections are retained for use by the replacement version.
+
+The upgrade process is triggered through the "adminFacet" of the instance, and requires specifying the new source bundle. (Note that a "null upgrade" that re-uses the original bundle is valid, and a legitimate approach to deleting accumulated state).
```

Comment:
Again, we should mention the need for a BundleCap somehow.

Although.. oh, huh. I'd forgotten what we decided about how Zoe manages the source of an upgrade. The *installation* has a specific BundleID, and the initial *instance* of that installation uses that bundle. But you upgrade the *instance*, and then it becomes.. detached from the installation?

If someone asks Zoe about the instance, does it tell you about the original installation, and then you might think that it's still running that old source code? Or can you ask it for the new BundleID? cc @erights to think about source-code access and auditability.

Anyways, if zoe upgrade takes a BundleID, we should add a sentence about how `controller.installBundle()` needs to happen first.

Reply:
I've just replaced "source bundle" with "source code" here for now, and I think the details (which are dictated by the specific [surface area of a contract instance's "adminFacet"](https://github.com/Agoric/agoric-sdk/blob/d99911630e32245e35f0a212d6d0c0e988abc052/packages/zoe/src/typeGuards.js#L344), cf. [packages/zoe/src/zoeService/startInstance.js](https://github.com/Agoric/agoric-sdk/blob/d99911630e32245e35f0a212d6d0c0e988abc052/packages/zoe/src/zoeService/startInstance.js#L179) ) can be clarified later by someone with authoritative information—and might turn out to be me anyway, depending upon how the near future goes. In the meantime, hopefully the example suffices.